TechnicalDocuments/Proposals/BabylonDesignLibrary.md

```diff
@@ -0,0 +1,208 @@
+we provide different styesheets based on the design.
+
+The idea behind this is to have reusable components with a predefined stylesheet displaying different data.
+The benefit of having a design library is
+1. to develop new UIs faster.
+2. to catch inconsistencies in the UI due to a design flaw or due to an engineering mistake.
+
+The requirement is to cover most of the common use cases and *not* to extend to edge cases.
+Also we must reuse our current tools like `Bento`.
+
+For instance, in our codebase a common UI element is a rounded button with background color. 
+For that we use `Component.Button` and a stylesheet while this approach is functional is has a few drawbacks.
+The attributes of the button (background color, alpha, corner radius etc) can either 
+1. have been redefined in another part of the codebase which is good from a design point of view *and if* we search
+for them we can use them, *if we don't* search for them we will just reimplement something which already exists.
+2. or the attributes of the button haven't been redefined in another part of the application and we may not
+notice that so a design flaw will be implemented. We are humans after all :).
+
+
+## Proposed solution
+We can introduce `DesignLibrary` which will be responsible for creating these components following
+the semantics of the design styleguide which will be provided to us.
+
+The purpose of  `DesignLibrary` isn't to expose the full potential of `Bento` components but to 
+follow specific design guidelines.
+As an example of this is: 
+A `Bento` component supports different behaviours like  `Deletable` the goal of this
+proposal isn't to replicate them or to make a wrapper around them but to follow specific design guidelines
+and to offer reusable UI elements.
+
+Also with `DesignLibrary` we want to limit the exposure of specific components, in order to be able to introduce  
+drastic design changes in the future as easy as possible.
+For example `DesignLibrary.ComponentsBuilder.button` returns `AnyRenderable` instead of `Component.Button`
+because in the future we may decide that we need to use a different button component. We want to depend on abstractions.
+
+Furthermore, `DesignLibrary` can also host alerts, custom viewcontroller animations, etc.
+
+``` swift
+
+struct DesignLibrary {
+    let tokens: Tokens
+
+    var viewControllerBuilder: ViewControllerBuilder {
+        return ViewControllerBuilder(tokens: Tokens)
+    }
+
+    var componentsBuilder: ComponentsBuilder {
+        return ComponentsBuilder(tokens: Tokens, traits: UITraitCollection())
+    }
+
+    init(
+        tokens: Tokens
+    ) {
+        self.tokens = tokens
+    }
+}
+
+extension DesignLibrary {
+    struct Tokens {
+        let typography: Typography
+        let colors: Colors
+    }
+}
+
+extension DesignLibrary.Tokens {
+    struct Typography {
+        struct FontAttributes {
```

Comment:
Can we use `UIFont.TextStyle` instead of `FontAttributes`? As far as I can tell it already contains information about the font's weight.

Alternatively we may define our own `UIFont.TextStyle`, to better match our design needs.